### PR TITLE
stability improvement: Android keystore migration

### PR DESCRIPTION
**Pull request by vzhovnitsky**

### CODE REVIEW

File: app/fragments/resolveOnboarding.ts [hunk 1]
```diff
@@ -19,6 +20,14 @@ export function resolveOnboarding(engine: Engine | null, isTestnet: boolean): On
         const address = getCurrentAddress();
         if (isAddressSecured(address.address, isTestnet)) {
             const passcodeSet = getPasscodeState() === PasscodeState.Set;
+            const isKeyStoreMigrated = storage.getBoolean(androidKeyStoreMigrated) ?? false;
+            const storageType = loadKeyStorageType();
+            const isKeyStore = storageType === 'key-store';
+
+            if (!isKeyStoreMigrated && Platform.OS === 'android' && isKeyStore) {
```
Comment (dvlkv) [inline app/fragments/resolveOnboarding.ts:27]: Simply - isKeyStore?

File: app/storage/secureStorage.ts [hunk 2]
```diff
@@ -120,6 +122,27 @@ async function doEncrypt(key: Buffer, data: Buffer) {
     return Buffer.concat([nonce, sealed]);
 }
 
+export async function migrateAndroidKeyStore(passcode?: string) {
+    const isMigrated = storage.getBoolean(androidKeyStoreMigrated);
```
Comment (dvlkv) [inline app/storage/secureStorage.ts:126]: Why do you need tag in storage?

File: app/storage/secureStorage.ts [hunk 3]
```diff
@@ -130,7 +153,7 @@ export async function encryptAndStoreAppKeyWithBiometrics(passcode: string) {
     }
 
     // Handle iOS
-    if (Platform.OS === 'ios') {
+    if (Platform.OS === 'ios' || Platform.OS === 'android') {
```
Comment (dvlkv) [inline app/storage/secureStorage.ts:156]: Remove if?

**Review by dvlkv — CHANGES_REQUESTED** [review]

File: app/storage/secureStorage.ts (hunk 2 above)
Comment (vzhovnitsky) [reply app/storage/secureStorage.ts:126]: just to be sure not to break anything

File: app/storage/secureStorage.ts (hunk 3 above)
Comment (vzhovnitsky) [reply app/storage/secureStorage.ts:156]: other platforms are in question here, but you are right if we build only for iOS and android this if statement is pointless

File: app/fragments/resolveOnboarding.ts (hunk 1 above)
Comment (vzhovnitsky) [reply app/fragments/resolveOnboarding.ts:27]: you are right

Comment (vzhovnitsky) [reply app/fragments/resolveOnboarding.ts:27]: I removed Platform deps.

**Review by dvlkv — APPROVED** [review]

**Pull request merged**